sql/core/src/test/scala/org/apache/spark/sql/MathExpressionsSuite.scala

```diff
@@ -70,11 +70,6 @@ class MathExpressionsSuite extends QueryTest {
         nnDoubleData.select(c('b)),
         (1 to 9).map(n => Row(f(n * -0.1))) :+ Row(Double.NegativeInfinity)
       )
-    } else {
-      checkAnswer(
-        nnDoubleData.select(c('b)),
-        (1 to 10).map(n => Row(null))
-      )
```

Comment:
Remove the test here since `NaN` cannot be compared with another, only test with `isNaN`, since we've check `NaN` in `MathFunctionSuite`, I think it's safe to delete this.